content/docs/installation/kubectl.md

```diff
@@ -50,17 +50,26 @@ $ kubectl create clusterrolebinding cluster-admin-binding \
 > and potentially broken states. Please ensure you follow the below steps when
 > uninstalling to prevent this happening.
 
-Before continuing, ensure that all cert-manager resources that have been created
+Before continuing, ensure that unwanted cert-manager resources that have been created
 by users have been deleted. You can check for any existing resources with the
 following command:
 
 ```bash
 $ kubectl get Issuers,ClusterIssuers,Certificates,CertificateRequests,Orders,Challenges --all-namespaces
 ```
+It is recommended that you delete all these resources before uninstalling cert-manager. 
+If plan on reinstalling later and don't want to lose some custom resources, you can keep them. 
+However, this can potentially lead to problems with finalizers. Some resources, like 
+`Challenges`, should be deleted to avoid [getting stuck in a pending state](#namespace-stuck-in-terminating-state). 
 
-Once all these resources have been deleted you are ready to uninstall
+Once the unneeded resources have been deleted, you are ready to uninstall
 cert-manager using the procedure determined by how you installed.
 
+> **Warning**: Uninstalling cert-manager or simply deleting a `Certificate` resource can result in 
+> TLS `Secret`s being deleted if they have `metadata.ownerReferences` set by cert-manager.
+> You can disable this behavior by setting `-enable-certificate-owner-ref=false` before deleting
```

Comment:
Suggestion: `-enable-certificate-owner-ref=false` will ensure that the owner reference is not added to a `Secret`, it does not disable the garbage collection as such. If the flag was set previously set to true and the user sets it to false, the owner references will be eventually removed, but this is only the case since cert-manager v1.8 (see https://github.com/cert-manager/cert-manager/pull/4888).
Perhaps we could add a little bit more detail to this warning- mention that `-enable-certificate-owner-ref=false` will ensure that owner refs are removed in cert-manager 1.8+ (I know that these docs are for 1.8 however I think it's possible that someone using an older version uninstalls by looking at the latest docs) and suggest that users check the owner references to ensure that they actually are removed. Also worth mentioning that `-enable-certificate-owner-ref` is set to false by default.